### PR TITLE
MAGECLOUD-3345: Add Node.JS Docker Image

### PR DESCRIPTION
**Pull request by bbatsche**
### Fixed Issues (if relevant)

[MAGECLOUD-3345](https://magento2.atlassian.net/browse/MAGECLOUD-3345)

### Manual testing scenarios

1. Run `php vendor/bin/ece-tools docker:build --node=8`
1. Verify `node` container has been added to `docker-compose.yml` with version `8`

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/ece-tools?pullRequest=453) <br/>All committers have signed the CLA.

**Review by billygilbert — APPROVED** [review]

File: src/Command/Docker/Build.php [hunk 1]
```diff
@@ -121,6 +123,11 @@ protected function configure()
                 null,
                 InputOption::VALUE_OPTIONAL,
                 'RabbitMQ version'
+            )->addOption(
+                self::OPTION_NODE,
+                null,
+                InputOption::VALUE_REQUIRED,
```
Comment (michaelcasey316) [inline src/Command/Docker/Build.php:149]: All of the other services have optional versions. Why does Node.js need a required version unlike the others?

**Review by michaelcasey316 — CHANGES_REQUESTED** [review]

File: src/Command/Docker/Build.php (hunk 1 above)
Comment (bbatsche) [reply src/Command/Docker/Build.php:149]: I believe having the other arguments' values optional is actually a bug.

For PHP, passing just `--php` with no value doesn't do anything different than omitting the flag altogether.

For other services, (`--es`, `--rmq`) passing the flag with no value does *not* add the service as one may expect. The service is only added to `docker-compose.yml` if the flag include a value.

Comment (michaelcasey316) [reply src/Command/Docker/Build.php:149]: Sounds good. Required makes sense then.

**Review by michaelcasey316 — APPROVED** [review]

**Comment by shiftedreality** [comment]
@bbatsche please resolve the conflicts

**Comment by andriyShevtsov** [comment]
QA approved

**Comment by shiftedreality** [comment]
Reopened in #476

**Pull request closed without merging**